src/neuroconv/tools/signal_processing.py

```diff
@@ -0,0 +1,56 @@
+"""Author: Cody Baker."""
+from typing import Optional
+
+import numpy as np
+
+
+def get_rising_frames_from_ttl(trace: np.ndarray, threshold: Optional[float] = None) -> np.ndarray:
+    """
+    Return the frame indices for rising events in a TTL pulse.
+
+    Parameters
+    ----------
+    trace: numpy.ndarray
+        A TTL pulse.
+    threshold: float, optional
+        The threshold used to distinguish on/off states in the trace.
+        The mean of the trace is used by default.
+
+    Returns
+    -------
+    rising_frames: numpy.ndarray
+        The frame indices of rising events.
+    """
+    threshold = threshold or np.mean(trace)
+
+    sign = np.sign(trace - threshold)
+    diff = np.diff(sign)
+    rising_frames = np.where(diff > 0)[0]
```

Comment:
```suggestion
    rising_frames = np.where(diff > 0)[0] + 1
```

I think you need to add one here. take a look at a simple trace e.g. `[-1, -1, 1]` to see